linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -6412,6 +6360,15 @@ private[optimizer] object OptimizerCore {
     }
   }
 
+  /** Makes a `Transient(Cast(expr, tpe))` but collapses consecutive casts. */
+  private def makeCast(expr: Tree, tpe: Type)(implicit pos: Position): Tree = {
+    val innerExpr = expr match {
+      case Transient(Cast(innerExpr, _)) => innerExpr
+      case _                             => expr
+    }
+    Transient(Cast(innerExpr, tpe))
```

Comment:
Feels like we should intersect the types here? I guess this isn't done because we do not have infra for it?

If yes, drop a comment?